contracts/interfaces/IAssessment.sol

```diff
@@ -3,180 +3,122 @@
 pragma solidity >=0.5.0;
 
 interface IAssessment {
-
-  /* ========== DATA STRUCTURES ========== */
-
-  struct Stake {
-    uint96 amount;
-    uint104 rewardsWithdrawableFromIndex;
-    uint16 fraudCount;
-    /*uint32 unused,*/
+  struct AssessmentData {
+    uint32 assessingGroupId;
+    uint32 cooldownPeriod;
   }
 
-  // Holds data for a vote belonging to an assessor.
-  //
-  // The structure is used to keep track of user's votes. Each vote is used to determine
-  // a user's share of rewards or to create a fraud resolution which excludes fraudulent votes
-  // from the initial poll.
-  struct Vote {
-    // Identifier of the claim or incident
-    uint80 assessmentId;
-
-    // If the assessor votes to accept the event it's true otherwise it's false
-    bool accepted;
+  struct AssessmentGroupView {
+    uint id;
+    bytes32 ipfsMetadata; // TODO: naming?
+    uint[] assessors;
+  }
 
-    // Date and time when the vote was cast
+  struct Ballot {
+    bytes32 ipfsHash;
+    Vote vote;
     uint32 timestamp;
-
-    // How many tokens were staked when the vote was cast
-    uint96 stakedAmount;
   }
 
-  // Holds poll results for an assessment.
-  //
-  // The structure is used to keep track of all votes on a given assessment such as how many NXM were
-  // used to cast accept and deny votes as well as when the poll started and when it ends.
-  struct Poll {
-    // The amount of NXM from accept votes
-    uint96 accepted;
-
-    // The amount of NXM from deny votes
-    uint96 denied;
-
-    // Timestamp of when the poll started.
+  struct Assessment {
+    uint32 assessorGroupId; // TODO: set AssessmentData instead
     uint32 start;
-
-    // Timestamp of when the poll ends.
-    uint32 end;
+    uint32 finalizedAt; // 0, not closed yet else timestamp of closure
+    uint8 acceptVotes; // 0, if not finalized yet, should only be set onced finalized
+    uint8 denyVotes; // 0, if not finalized yet, should only be set onced finalized
+    mapping(uint assessor => Ballot) ballot; // only stores latest choice
```

Comment:
Do not nest mappings inside structs, there will be issues with reading and writing to it, for example cannot write a `Assessment memory` to storage due to the mapping, likewise reading from storage to memory could be misleading. Define a `mapping(uint => mapping(uint => Ballot))` instead.